### PR TITLE
add chemical datatypes

### PR DESCRIPTION
**Pull request by bgruening**
This will add a bunch of molecule datatypes to Galaxy. It was developed as part of the [ChemicalToolBox](https://github.com/bgruening/galaxytools/tree/master/chemicaltoolbox) and I feel it's time to move them into core.

The converters rely on a binary called OpenBabel which can will be installed during runtime, if conda dependency resolution is enabled.

I marked it as WIP as I have some smaller corrections to do, but would appreciate a first review and it it's ok to move them into main.

### CODE REVIEW

File: config/datatypes_conf.xml.sample [hunk 1]
```diff
@@ -484,6 +531,13 @@
     <sniffer type="galaxy.datatypes.proteomics:Msp"/>
     <sniffer type="galaxy.datatypes.proteomics:SPLib"/>
     <sniffer type="galaxy.datatypes.proteomics:ThermoRAW"/>
+    <sniffer type="galaxy.datatypes.molecules:SDF"/>
+    <sniffer type="galaxy.datatypes.molecules:PDB"/>
+    <sniffer type="galaxy.datatypes.molecules:MOL2"/>
+    <sniffer type="galaxy.datatypes.molecules:InChI"/>
+    <sniffer type="galaxy.datatypes.molecules:FPS"/>
+    <sniffer type="galaxy.datatypes.molecules:CML"/>
+    <!-- TODO: see molecules.py <sniffer type="galaxy.datatypes.molecules:SMILES"/>-->
```
Comment (jmchilton) [inline config/datatypes_conf.xml.sample]: Can you update the doctests for `guess_ext` with sniffer tests for these datatypes - at least the ones you have test data in the repo for:

https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/datatypes/sniff.py#L259

**Comment by jmchilton** [comment]
Conda is paying dividends all over the place, this is awesome @bgruening.

File: config/datatypes_conf.xml.sample (hunk 1 above)
Comment (bgruening) [inline config/datatypes_conf.xml.sample]: Actually, I already have some doctests. But I need to extend them and add more test data. Still struggling to find small once. Will do so!

Comment (jmchilton) [inline config/datatypes_conf.xml.sample]: Your doctests are testing the sniff method - putting them up in `guess_ext` would test the actual datatype conf file - that the sniffer is in the right order, etc...

**Comment by bgruening** [comment]
@jmchilton can you please look over it. I'm not sure this all is needed. Why do I need to edit registry.py?

**Comment by martenson** [comment]
@bgruening Does lib/galaxy/datatypes/test/2zbz.pdb have to have 4k lines?
Besides that it looks very fine to me.

**Comment by bgruening** [comment]
@martenson I can search for a smaller file. Do you know why I need to touch the registry.py? Shouldn’t be the ordering in datatypes_conf be sufficient?

**Comment by martenson** [comment]
@bgruening I thought the conf is sufficient. Why do you think you need to touch registry.py?

**Comment by bgruening** [comment]
@martenson I need to have these changes to make tests pass: https://github.com/galaxyproject/galaxy/pull/1941/files#diff-81f0f003556ce7dde04d755ff6283680

**Comment by martenson** [comment]
@bgruening I cannot comment on what the best way is but I see no harm done by adding it there.

**Comment by bgruening** [comment]
@martenson I have found a smaller file and rebased this PR.

**Comment by bgruening** [comment]
@galaxybot test this.

**Comment by jmchilton** [comment]
The registry.py changes should not be needed, does this fail without them?

**Comment by martenson** [comment]
@jmchilton I think that is what @bgruening said

**Comment by jmchilton** [comment]
Okay - I see that now - I'll pull this down and hack on it.

**Comment by jmchilton** [comment]
Clearly I hadn't made the change to test the datatypes conf that I thought I did - I've included the fix with https://github.com/bgruening/galaxy/pull/9. I wonder if this is the problem with #1156...

**Comment by bgruening** [comment]
Thanks @jmchilton! Seems to work as expected! Can we merge this and trigger a new test run in #1156.

**Comment by bgruening** [comment]
Thanks for the merge and the review and fixing and for everything! :)

**Pull request merged**